<a href="https://colab.research.google.com/github/ProgramCrafter/colab-foxford-bigdata/blob/master/%D0%9B%D0%B5%D0%BA%D1%86%D0%B8%D1%8F_%E2%84%961.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/98155/upd1.jpg"></p>

___

<font color="#000000">Основы Python и глубокого обучения</font>
<font color="#AEBAC5"> → </font>
<font color="#AEBAC5">Компьютерное зрение (CV)</font>
<font color="#AEBAC5"> → </font>
<font color="#AEBAC5">Обработка естественного языка (NLP)</font>

___

<h1><strong><font color="#FF732B">Кто такие эти ваши ноутбуки?</font></strong></h1>

<h3><strong><font color="#809BFF">Задача №1</font></strong></h3>

Ниже расположен код на языке Python, который отвечает за вывод информации на экран. Перейди в эту ячейку и нажми `Command/Ctrl` + `Enter` на клавиатуре или кликни на кнопку Play слева от строчки кода (самый первый вызов — не моментальный и требует 3-4 секунды, это нормально).

In [1]:
# выполни код в ячейке
print('Отлично! Не так уж и сложно.')

Отлично! Не так уж и сложно.


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/98336/upd5.png"></p>

<h1><strong><font color="#FF732B">Знакомимся с первыми данными</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92609/3art.png"></p>

| Стоимость, млн $ | Название | Художник | Год создания | Аукционный дом |
--- | --- | --- | --- | ---
| 193.5 | Алжирские женщины («Версия O») | Пабло Пикассо 🇪🇸 | 1955 | Christie's |
| 170.1 | Сон | Пабло Пикассо 🇪🇸 | 1932 | Другой |
| 161.4 | Портрет доктора Гаше | Винсент Ван Гог 🇳🇱 | 1890 | Christie's |
| 141 | Мальчик с трубкой | Пабло Пикассо 🇪🇸 | 1905 | Sotheby's |
| 124.8 | Обнажённая, зелёные листья и бюст | Пабло Пикассо 🇪🇸 | 1932 | Christie's |
| 121.3 | Ирисы | Винсент Ван Гог 🇳🇱 | 1889 | Sotheby's |
| 120.8 | Дора Маар с кошкой | Пабло Пикассо 🇪🇸 | 1941 | Sotheby's |
| 117.1 | Молодая девушка с цветочной корзиной | Пабло Пикассо 🇪🇸 | 1905 | Christie's |
| 115.7 | Авария серебряной машины (Двойная катастрофа) | Энди Уорхол 🇺🇸 | 1963 | Sotheby's |
| 112.2 | Автопортрет без бороды | Винсент Ван Гог 🇳🇱 | 1889 | Christie's |
| 101.7 | Свадьба Пьеретты | Пабло Пикассо 🇪🇸 | 1905 | Другой |
| 98.7 | Я, Пикассо | Пабло Пикассо 🇪🇸 | 1901 | Sotheby's |
| 89.3 | Подсолнухи | Винсент Ван Гог 🇳🇱 | 1889 | Christie's |
| 88.5 | Тройной Элвис | Энди Уорхол 🇺🇸 | 1963 | Christie's |
| 88.4 | Авария зелёной машины | Энди Уорхол 🇺🇸 | 1963 | Christie's |
| 84.8 | Огороженное поле с пахарем | Винсент Ван Гог 🇳🇱 | 1889 | Christie's |
| 83.9 | Проворный кролик | Пабло Пикассо 🇪🇸 | 1904 | Sotheby's |
| 83.3 | Акробат и молодой Арлекин | Пабло Пикассо 🇪🇸 | 1905 | Christie's |
| 81.7 | Женщина со скрещёнными руками | Пабло Пикассо 🇪🇸 | 1902 | Christie's |
| 76 | Женщина, сидящая в саду | Пабло Пикассо 🇪🇸 | 1938 | Sotheby's |
| 75.7 | Молодая крестьянка в соломенной шляпке, сидящая в пшенице | Винсент Ван Гог 🇳🇱 | 1890 | Другой |
| 75.2 | Четыре Марлона | Энди Уорхол 🇺🇸 | 1966 | Christie's |
| 74.3 | Мужчины в ее жизни | Энди Уорхол 🇺🇸 | 1962 | Другой |
| 72.8 | Певица кабаре | Пабло Пикассо 🇪🇸 | 1901 | Sotheby's |
| 72.7 | Бюст женщины (Дора Маар) | Пабло Пикассо 🇪🇸 | 1938 | Christie's |
| 71.5 | Аллея Алискана | Винсент Ван Гог 🇳🇱 | 1888 | Sotheby's |
| 70.5 | Женщина в берете и клетчатом платье (Мария Тереза Вальтер) | Пабло Пикассо 🇪🇸 | 1937 | Sotheby's |
| | | | | |
| | | | | *Таблица 1.0* |

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92612/4report.png"></p>

<h1><strong><font color="#FF732B">Складываем, вычитаем, умножаем и делим</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92616/5oper.png"></p>

<h3><strong><font color="#809BFF">Задача №2</font></strong></h3>

Вычисли разницу в датах создания картин «Алжирские женщины» и «Обнажённая, зелёные листья и бюст» Пабло Пикассо. Подставь значения и выполни код.

In [2]:
# подставь значения вместо многоточий
print(1955 - 1932)

23


<h3><strong><font color="#809BFF">Задача №3</font></strong></h3>

Вычисли общую стоимость картин «Авария серебряной машины (Двойная катастрофа)», «Тройной Элвис» и «Мужчины в ее жизни» Энди Уорхола. Напиши выражение и выполни код.

In [1]:
# напиши выражение
print(115.7 + 88.5 + 74.3)

278.5


<h3><strong><font color="#809BFF">Задача №4</font></strong></h3>

Вычисли, насколько 3 самые дорогие картины Пабло Пикассо дороже 3-х самых дорогих картин Винсента Ван Гога. Подставь значения и выполни код.

In [2]:
# напиши выражение

print(193.5 + 170.1 - 161.4 + 141 - 121.3 - 112.2)

109.70000000000003


<h3><strong><font color="#809BFF">Задача №5</font></strong></h3>

Вычисли, насколько 2 самые дорогие картины, проданные аукционным домом Christie's дороже 2-х самых дорогих картин, проданных Sotheby's. Напиши выражение и выполни код.

In [3]:
print(193.5 + 161.4 - 141 - 121.3)

92.59999999999998


<h3><strong><font color="#809BFF">Задача №6</font></strong></h3>

Вычисли, насколько доля картин Винсента Ван Гога и Энди Уорхола, проданных аукционными домами Christie's и Sotheby's, меньше доли картин всех остальных художников в топ-100, проданных всеми аукционными домами. Будь внимателен, беря отдельные выражения в скобки. Напечатай результат на экране.

In [4]:
E = (193.5 + 170.1 + 161.4 + 141.0 + 124.8 + 121.3 +
     120.8 + 117.1 + 115.7 + 112.2 + 101.7 + 98.7  +
     89.3  + 88.5  + 88.4  + 84.8  + 83.9  + 83.3  +
     81.7  + 76.0  + 75.7  + 75.2  + 74.3  + 72.8  +
     72.7  + 71.5  + 70.5)

print((E -
       (161.4 + 121.3 + 115.7 + 112.2 + 89.3 +
        88.5  + 88.4  + 84.8  + 75.7  + 75.2 + 74.3)) / E -
      (161.4 + 121.3 + 115.7 + 112.2 + 89.3 +
       88.5  + 88.4  + 84.8  + 75.2) / E)

0.2686399942173551


<h1><strong><font color="#FF732B">Сохраняем нужные значения</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92618/6var.png"></p>

<h3><strong><font color="#809BFF">Задача №7</font></strong></h3>

Узнай продолжительность жизни каждого художника и сохрани соответствующие значения в переменные. Напечатай результат для одного из художников.

In [5]:
# подставь нужное значение вместо многоточия
picasso_lifetime = 91

# объяви переменную для Винсента Ван Гога и сохраните в нее значение
van_gogh_lifetime = 37

# переменная и значение для Энди Уорхола
warhol_lifetime = 58

# напечатай результат для Винсента Ван Гога
print(van_gogh_lifetime)

37


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92620/7repo.png"></p>

<h3><strong><font color="#809BFF">Задача №8</font></strong></h3>

Посчитай, сколько картин каждого художника находится в списке топ-100. Общее количество картин всех 3-х художников — также сохрани в отдельную переменную и напечатай это значение на экране.

In [6]:
# подставь нужное значение вместо троеточия
picasso_top_100_amount = 15

# объяви переменную для Винсента Ван Гога и сохрани в нее значение
van_gogh_top_100_amount = 7

# переменная и значение для Энди Уорхола
warhol_top_100_amount = 5

# общее количество картин
total_top_100_amount = picasso_top_100_amount + van_gogh_top_100_amount + warhol_top_100_amount

# напечатай общее количество картин
print(total_top_100_amount)

27


<h3><strong><font color="#809BFF">Задача №9</font></strong></h3>

Для каждого художника вычисли разницу между самой поздней и самой ранней картиной в списке топ-100. На экране напечатай результат для Пабло Пикассо. Подумай над получившимися значениями — о чем они могут говорить?

In [7]:
# подставь нужное значение вместо многоточия
picasso_creation_range = 1955 - 1901

# переменная и значение для Винсента Ван Гога
van_gogh_creation_range = 1890 - 1888

# переменная и значение для Энди Уорхола
warhol_creation_range = 1966 - 1962

# напечатай результат для Пабло Пикассо
print(picasso_creation_range)

54


<h3><strong><font color="#809BFF">Задача №10</font></strong></h3>

Вычисли долю картин каждого художника среди общего количества. Все значения, которые нужны для решения задачи, уже сохранены в свои переменные — используй их. На экране напечатай результат для Винсента Ван Гога.

In [15]:
# доля Пабло Пикассо
picasso_share = (picasso_top_100_amount / total_top_100_amount)

# доля Винсента Ван Гога
van_gogh_share = (van_gogh_top_100_amount / total_top_100_amount)

# доля Энди Уорхола
warhol_share = (warhol_top_100_amount / total_top_100_amount)

# напечатай долю Винсента Ван Гога
print(van_gogh_share)

0.25925925925925924


<h1><strong><font color="#FF732B">Смотрим на промежуточный результат</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/92622/8repo.png"></p>

<h1><strong><font color="#FC4262">Домашняя работа</font></strong></h1>

Вот несколько шагов, чтобы настроить удобную работу над домашними задачами:

1. В сервисе [Google Drive](https://www.google.com/intl/ru/drive/) создай папку `foxford_ds_study`, в которой ты будешь хранить все учебные материалы с этого курса.
2. Сохрани копию этой тетрадки `Лекция №1.ipynb` *(File → Save a copy in Drive)* в папку `foxford_ds_study`.

Зачем это нужно? 

Файл, в котором работает преподаватель — это «оригинальный» файл и мы не можем работать в нем все вместе, ведь у каждого свой код! Каждому ученику необходимо сделать свою копию файла и выполнять домашку именно в ней. Если хочешь, ты можешь назвать свой файл по имени, например, `Лекция №1 Иван Кнопочкин.ipynb`.

В начале следующей лекции мы обсудим эти задачи вместе.

<h3><strong><font color="#809BFF">Задача №11</font></strong></h3>

Вычисли, насколько доля картин Пабло Пикассо и Энди Уорхола в топ-100 больше доли картин Винсента Ван Гога и выведи результат на экран.

In [20]:
# напиши выражение
print(picasso_share + warhol_share - van_gogh_share)

print('HOW 0.13? o_O')

0.48148148148148145
HOW 0.13? o_O


<div>
  <details>
    <summary>Подсказка 1/1</summary>
    <p>Найди сумму долей Пабло Пикассо и Энди Уорхола и вычти из нее долю Винсента Ван Гога.</p>
  </details>
</div>

<div>
  <details>
    <summary>Результат</summary>
    <p>0.13</p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №12</font></strong></h3>

Обозначим картины с низкой стоимостью как те, цена которых менее 75 млн $. Посчитай среднюю стоимость таких картин и напечатай результат на экране.

In [12]:
# кол-во картин
least_price_amount = 5

# общая стоимость
least_price_cost = 74.3 + 72.8 + 72.7 + 71.5 + 70.5

# средняя стоимость
least_price_avg_price = least_price_cost / least_price_amount

# напечатай результат
print(least_price_avg_price)

72.36


<div>
  <details>
	  <summary>Подсказка 1/1</summary>
    <p>Среднее значение — это число, равное сумме всех чисел множества, делённое на их количество.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>72.36</p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №13</font></strong></h3>

Обозначим картины с высокой стоимостью как те, цена которых более 124 млн $. Посчитай среднюю стоимость таких картин и напечатай результат на экране.

In [13]:
most_price_amount = 5
most_price_cost = 193.5 + 170.1 + 161.4 + 141 + 124.8
most_price_avg_price = most_price_cost / most_price_amount

print(most_price_avg_price)

158.16


<div>
  <details>
	  <summary>Подсказка 1/1</summary>
    <p>Среднее значение — это число, равное сумме всех чисел множества, делённое на их количество.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>158.16</p>
  </details>
</div>

<h3><strong><font color="#80AF4B">Отправка решения</font></strong></h3>

In [23]:
!git add *1.ipynb
!git commit
!git push

Aborting commit due to empty commit message.
Everything up-to-date
